wordpress-mobile/WordPress-Android
# Refactor PostLikeUseCase

Parent issue #12028

This PR refactors the PostLikeUseCase
- to take advantage of flow
- integrated into the legacy ReaderPostListFragment. 

Implementation Details
- ReaderPostActions
    - Added two new Like action method, basically separates the single method into two. 
        `performLikeActionLocal`  and `performActionRemote`.   
        `ReaderDetailFragment` currently uses `performLikeAction()`, so the existing method calls the two new methods. We will address ReaderDetailFragment in a future phase of the RI project.
    - Removed ReaderEvents that were no longer needed
- ReaderPostActionsWrapper
    - Add wrappers around the two new ReaderPostActions methods listed above
- ReaderPostListViewModel
    - Add `onLikeButtonClicked()` to send the Like click through `ReaderPostActionHandler`
- ReaderPostAdapter
    - Removed toggleLike method
    - Use `mOnPostListItemButtonListener` for Like clicks
- Updated `ReaderPostCardsActionHandler`
    - Uses `PostLikeState`  instead of ReaderRepositoryCommunication
    - Include handling UpdateInLocalDB and posting to _refreshPosts
- PostLikeUseCase
    - Changed to use flows so we can handle localDb updates and refresh legacy reader tabs
    - Replaced ReaderRepositoryCommunication with PostLikeState 
    - Removed references to eventbus
- ReaderPostListFragment
    - Hooked up the Like case in `onButtonClicked`
- ReaderRepositoryEvent
    - Removed unused events
- ReaderLikeUseCaseTest
    - Fixed tests

**NOTE**: This PR does not include any PostLikeUseCase tests (I commented out existing ones), as they need to be rewritten to work with flow. The tests will follow behind shortly after I research how to test with flows.

To test:
1. Open the app with FF on
2. Navigate to the following tab
3. Like a post
4. The icon color will change and the count goes up
5. Unlike a post
6. The icon color will no longer be colored and the count goes down
7. Navigate to the Discover tab
8. Repeat steps 3-6

Regression test post detail
1. Open the app with FF on
2. Navigate to post detail
3. Like a post
4. The icon color will change and the count goes up
5. Unlike a post
6. The icon color will no longer be colored and the count goes down
PR submission checklist:
